codec-http2/src/main/java/io/netty/handler/codec/http2/HttpToHttp2ConnectionHandler.java

```diff
@@ -116,4 +120,22 @@ public void write(ChannelHandlerContext ctx, Object msg, ChannelPromise promise)
             promiseAggregator.doneAllocatingPromises();
         }
     }
+
+    private static void writeHeaders(ChannelHandlerContext ctx, Http2ConnectionEncoder encoder, int streamId,
+                                     HttpHeaders headers, Http2Headers http2Headers, boolean endStream,
+                                     SimpleChannelPromiseAggregator promiseAggregator) {
+        int dependencyId = headers.getInt(
+                HttpConversionUtil.ExtensionHeaderNames.STREAM_DEPENDENCY_ID.text(), -1);
```

Comment:
yeah that's what I was thinking.  This could boil down to something like this:

```java
int dependencyId = headers.getInt(
                HttpConversionUtil.ExtensionHeaderNames.STREAM_DEPENDENCY_ID.text(), 0);
short weight = headers.getShort(
                    HttpConversionUtil.ExtensionHeaderNames.STREAM_WEIGHT.text(), Http2CodecUtil.DEFAULT_PRIORITY_WEIGHT);
encoder.writeHeaders(ctx, streamId, http2Headers, dependencyId, weight, false,
                    0, endStream, promiseAggregator.newPromise());
```